tests/Unit/Helpers/NumericalAlgorithms/FiniteDifference/Exact.hpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Evolution/DgSubcell/SliceData.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace TestHelpers::fd::reconstruction {
+/*!
+ * \brief Test that the reconstruction procedure can exactly reconstruct
+ * polynomials of degree equal to the degree of the basis.
+ *
+ * For example, for a minmod, MC, or other second-order TVD limiter, we must be
+ * able to reconstruct linear functions exactly. For a third-order WENO scheme,
+ * we must be able to reconstruct quadratic functions exactly. This is done
+ * by setting up the function
+ *
+ * \f{align}{
+ *  u=\sum_{i=1}^p x^p + y^p + z^p
+ * \f}
+ *
+ * where \f$x,y,z\f$ are compute by the logical coordinates translated by
+ * \f$4\f$ in \f$y\f$ and by \f$8\f$ in \f$z\f$. The translation is done
+ * to catch subtle bugs that may arise from indexing in the wrong dimension.
+ */
+template <size_t Dim, typename F>
+void test_reconstruction_is_exact_if_in_basis(const size_t max_degree,
+                                              const size_t points_per_dimension,
+                                              const size_t stencil_width,
+                                              const F& invoke_recons) {
+  const size_t number_of_vars = 2;  // arbitrary, 2 is "cheap but not trivial"
+
+  const Mesh<Dim> mesh{points_per_dimension, Spectral::Basis::FiniteDifference,
+                       Spectral::Quadrature::CellCentered};
+  auto log_coords = logical_coordinates(mesh);
+  // Make the logical coordinates different in each direction
+  for (size_t i = 1; i < Dim; ++i) {
+    log_coords.get(i) = log_coords.get(i) + 4.0 * i;
```

Comment:
This doesn't match the translation described in the documentation for the x direction case

Reply:
Oh I think I misread the documentation on this 🤦